src/metrics.cpp

```diff
@@ -231,6 +263,20 @@ int printStats(bool mining)
     } else {
         std::cout << "           " << _("Block height") << " | " << height << std::endl;
     }
+
+    const Consensus::Params& params = Params().GetConsensus();
+    int next_upgrade_height = NextActivationHeight(height, params).get_value_or(0);
+    int next_upgrade = NextEpoch(height, params).get_value_or(0);
+    int64_t seconds_left = (next_upgrade_height - height) * params.PoWTargetSpacing(next_upgrade_height);
```

Comment:
This seems wrong because it uses the target spacing *at* the upgrade height, not *up to* the upgrade height.

```suggestion
    int64_t seconds_left = (next_upgrade_height - height) * params.PoWTargetSpacing(next_upgrade_height - 1);
```